### ISSUE TITLE
erase-install hangs when prompted to download a build with an invalid id

### ISSUE THREAD

**Issue by aschwanb** [grahampugh/erase-install#356]
**Describe the bug**
erase-install hangs when prompted to download a build with an invalid id.

**To Reproduce**
In the terminal:
`sudo /Library/Management/erase-install/erase-install.sh --build=no_build_with_thisID`
 
**Expected behavior**
erase-install does exit with an error message to the extend of "No build with this ID available"

**Code/log output**
```zsh
% sudo /Library/Management/erase-install/erase-install.sh --build=no_build_with_thisID
Password:

2023-02-15 15:38:38  [erase-install] v28.1 script execution started: Wed Feb 15 15:38:38 CET 2023
2023-02-15 15:38:38  [erase-install] System version: 12.6.3 (Build: 21G419)
2023-02-15 15:38:38  [check_for_dialog_app] dialog is installed (/Library/Application Support/Dialog/Dialog.app)
2023-02-15 15:38:38  [erase-install] Caffeinating this script (pid=16991)
2023-02-15 15:38:38  [erase-install] Looking for existing installer app or pkg
2023-02-15 15:38:38  [find_existing_installer] No valid installer found.
2023-02-15 15:38:38  [erase-install] Checking if the cached installer matches requested build...
2023-02-15 15:38:38  [compare_build_versions] Comparing (1) no_build_with_thisID with (2) 
2023-02-15 15:38:38  [compare_build_versions] no_build_with_thisID > 
2023-02-15 15:38:38  [erase-install] Existing installer does not match requested build, so replacing...
2023-02-15 15:38:38  [overwrite_existing_installer] Overwrite option selected. Deleting existing version.
2023-02-15 15:38:38  [erase-install] Running on architecture i386
2023-02-15 15:38:38  [check_for_mist] mist is installed (/usr/local/bin/mist)
┌──────────────────┐
│ INPUT VALIDATION │
└──────────────────┘
  ├─ Search only for latest (first) result will be 'false'...
  ├─ Include betas in search results will be 'false'...
  ├─ Only include compatible installers will be 'true'...
  ├─ Export path will be '/Library/Management/erase-install/mist-list.json'...
  ├─ Export path file extension is valid...
  ├─ Output type will be 'ascii'...
┌────────┐
│ SEARCH │
└────────┘
  ├─ Searching for macOS Installer versions...
  ├─ Exported list as JSON: '/Library/Management/erase-install/mist-list.json'
  └─ Found 21 macOS Installer(s) available for download

Identifier │ Name           │ Version │ Build   │ Size     │ Date       │ Compatible
───────────┼────────────────┼─────────┼─────────┼──────────┼────────────┼───────────
032-48342  │ macOS Ventura  │ 13.2.1  │ 22D68   │ 12.56 GB │ 2023-02-13 │ True      
032-35706  │ macOS Ventura  │ 13.2    │ 22D49   │ 12.56 GB │ 2023-01-23 │ True      
012-60271  │ macOS Ventura  │ 13.1    │ 22C65   │ 12.23 GB │ 2023-01-11 │ True      
012-93766  │ macOS Ventura  │ 13.0.1  │ 22A400  │ 12.16 GB │ 2022-11-09 │ True      
032-33828  │ macOS Monterey │ 12.6.3  │ 21G419  │ 12.41 GB │ 2023-01-31 │ True      
032-12843  │ macOS Monterey │ 12.6.2  │ 21G320  │ 12.40 GB │ 2023-01-11 │ True      
012-90254  │ macOS Monterey │ 12.6.1  │ 21G217  │ 12.41 GB │ 2022-10-24 │ True      
032-33756  │ macOS Big Sur  │ 11.7.3  │ 20G1116 │ 12.42 GB │ 2023-01-31 │ True      
032-12833  │ macOS Big Sur  │ 11.7.2  │ 20G1020 │ 12.42 GB │ 2023-01-11 │ True      
012-90253  │ macOS Big Sur  │ 11.7.1  │ 20G918  │ 12.42 GB │ 2022-10-24 │ True      
012-08272  │ macOS Big Sur  │ 11.6.6  │ 20G624  │ 12.42 GB │ 2022-05-24 │ True      
002-77154  │ macOS Big Sur  │ 11.6.5  │ 20G527  │ 12.42 GB │ 2022-04-11 │ True      
002-65695  │ macOS Big Sur  │ 11.6.4  │ 20G417  │ 12.45 GB │ 2022-02-17 │ True      
002-57023  │ macOS Big Sur  │ 11.6.3  │ 20G415  │ 12.44 GB │ 2022-01-26 │ True      
002-42341  │ macOS Big Sur  │ 11.6.2  │ 20G314  │ 12.44 GB │ 2022-01-14 │ True      
002-23589  │ macOS Big Sur  │ 11.6.1  │ 20G224  │ 12.43 GB │ 2021-12-01 │ True      
071-78704  │ macOS Big Sur  │ 11.5.2  │ 20G95   │ 12.45 GB │ 2021-08-18 │ True      
001-68446  │ macOS Catalina │ 10.15.7 │ 19H15   │ 08.75 GB │ 2020-11-11 │ True      
001-51042  │ macOS Catalina │ 10.15.7 │ 19H2    │ 08.75 GB │ 2020-09-24 │ True      
001-36801  │ macOS Catalina │ 10.15.6 │ 19G2021 │ 08.75 GB │ 2020-08-12 │ True      
061-26589  │ macOS Mojave   │ 10.14.6 │ 18G103  │ 06.52 GB │ 2019-10-14 │ True      
```

**Screenshots**
N/A

**Environment (please complete the following information):**
 - OS version
 - macOS 13.2 (ARM)
 - macOS 12.6.3 (Intel)
 - erase-install version
 - v28.1

**Additional context**
```zsh
% mist --version
1.10 (Latest: 1.10)
```

**Comment by grahampugh** [comment]
I cannot reproduce this. In my tests it errors out correctly.

```
sudo ./erase-install.sh --build WEFSFF            

2023-02-27 20:37:44  [erase-install] v29.1 script execution started: Mon 27 Feb 2023 20:37:44 CET
2023-02-27 20:37:44  [erase-install] System version: 13.2 (Build: 22D49)
2023-02-27 20:37:44  [check_for_dialog_app] dialog is installed (/Library/Application Support/Dialog/Dialog.app)
2023-02-27 20:37:44  [check_for_dialog_app] Creating dialog log (/var/tmp/dialog.3h2)...
2023-02-27 20:37:44  [erase-install] Caffeinating this script (pid=96617)
2023-02-27 20:37:44  [erase-install] Looking for existing installer app or pkg
2023-02-27 20:37:44  [find_existing_installer] No valid installer found.
2023-02-27 20:37:44  [erase-install] Checking if the cached installer matches requested build...
2023-02-27 20:37:44  [compare_build_versions] Comparing (1) WEFSFF with (2) 
2023-02-27 20:37:44  [compare_build_versions] WEFSFF > 
2023-02-27 20:37:44  [erase-install] Existing installer does not match requested build, so replacing...
2023-02-27 20:37:44  [overwrite_existing_installer] Overwrite option selected. Deleting existing version.
2023-02-27 20:37:44  [erase-install] Running on architecture arm64
2023-02-27 20:37:44  [check_for_mist] mist is installed (/usr/local/bin/mist)
┌──────────────────┐
│ INPUT VALIDATION │
└──────────────────┘
  ├─ Search only for latest (first) result will be 'false'...
  ├─ Include betas in search results will be 'false'...
  ├─ Only include compatible installers will be 'true'...
  ├─ Export path will be '/Library/Management/erase-install/mist-list.json'...
  ├─ Export path file extension is valid...
  ├─ Output type will be 'ascii'...
┌────────┐
│ SEARCH │
└────────┘
  ├─ Searching for macOS Installer versions...
  ├─ Exported list as JSON: '/Library/Management/erase-install/mist-list.json'
  └─ Found 18 macOS Installer(s) available for download

Identifier │ Name           │ Version │ Build   │ Size     │ Date       │ Compatible
───────────┼────────────────┼─────────┼─────────┼──────────┼────────────┼───────────
032-48342  │ macOS Ventura  │ 13.2.1  │ 22D68   │ 12.56 GB │ 2023-02-20 │ True      
032-35706  │ macOS Ventura  │ 13.2    │ 22D49   │ 12.56 GB │ 2023-01-23 │ True      
012-60271  │ macOS Ventura  │ 13.1    │ 22C65   │ 12.23 GB │ 2023-01-11 │ True      
012-93766  │ macOS Ventura  │ 13.0.1  │ 22A400  │ 12.16 GB │ 2022-11-09 │ True      
032-33828  │ macOS Monterey │ 12.6.3  │ 21G419  │ 12.41 GB │ 2023-01-31 │ True      
032-12843  │ macOS Monterey │ 12.6.2  │ 21G320  │ 12.40 GB │ 2023-01-11 │ True      
012-90254  │ macOS Monterey │ 12.6.1  │ 21G217  │ 12.41 GB │ 2022-10-24 │ True      
032-50519  │ macOS Big Sur  │ 11.7.4  │ 20G1120 │ 12.42 GB │ 2023-02-15 │ True      
032-33756  │ macOS Big Sur  │ 11.7.3  │ 20G1116 │ 12.42 GB │ 2023-01-31 │ True      
032-12833  │ macOS Big Sur  │ 11.7.2  │ 20G1020 │ 12.42 GB │ 2023-01-11 │ True      
012-90253  │ macOS Big Sur  │ 11.7.1  │ 20G918  │ 12.42 GB │ 2022-10-24 │ True      
012-08272  │ macOS Big Sur  │ 11.6.6  │ 20G624  │ 12.42 GB │ 2022-05-24 │ True      
002-77154  │ macOS Big Sur  │ 11.6.5  │ 20G527  │ 12.42 GB │ 2022-04-11 │ True      
002-65695  │ macOS Big Sur  │ 11.6.4  │ 20G417  │ 12.45 GB │ 2022-02-17 │ True      
002-57023  │ macOS Big Sur  │ 11.6.3  │ 20G415  │ 12.44 GB │ 2022-01-26 │ True      
002-42341  │ macOS Big Sur  │ 11.6.2  │ 20G314  │ 12.44 GB │ 2022-01-14 │ True      
002-23589  │ macOS Big Sur  │ 11.6.1  │ 20G224  │ 12.43 GB │ 2021-12-01 │ True      
071-78704  │ macOS Big Sur  │ 11.5.2  │ 20G95   │ 12.45 GB │ 2021-08-18 │ True      

2023-02-27 20:37:47  [run_mist] ERROR: build is not available


2023-02-27 20:37:47  [erase-install] attempting to terminate the 'caffeinate' process - Termination message indicates success

2023-02-27 20:37:48  [finish] sending quit message to dialog (/var/tmp/dialog.3h2)
2023-02-27 20:37:48  [finish] Script exit code: 1
```

**Comment by aschwanb** [comment]
Can confirm. This is fixed in `v29.1`.

Thank you :-)
